miniwin/miniwin/src/miniwin_ddraw.cpp

```diff
@@ -26,18 +57,51 @@ struct DirectDrawSurfaceImpl : public IDirectDrawSurface3 {
 		LPDDBLTFX lpDDBltFx
 	) override
 	{
+		if (!renderer) {
+			return DDERR_GENERIC;
+		}
+		SDL_FRect srcRect = ConvertRect(lpSrcRect);
+		SDL_FRect dstRect = ConvertRect(lpDestRect);
+		SDL_RenderTexture(renderer, static_cast<DirectDrawSurfaceImpl*>(lpDDSrcSurface)->texture, &srcRect, &dstRect);
+		SDL_RenderPresent(renderer);
 		return DD_OK;
 	}
 	HRESULT BltFast(DWORD dwX, DWORD dwY, LPDIRECTDRAWSURFACE lpDDSrcSurface, LPRECT lpSrcRect, DDBltFastFlags dwTrans)
 		override
 	{
+		if (!renderer) {
+			return DDERR_GENERIC;
+		}
+		SDL_FRect dstRect = {
+			(float) dwX,
+			(float) dwY,
+			(float) (lpSrcRect->right - lpSrcRect->left),
+			(float) (lpSrcRect->bottom - lpSrcRect->top)
+		};
+		SDL_FRect srcRect = ConvertRect(lpSrcRect);
+		SDL_RenderTexture(renderer, static_cast<DirectDrawSurfaceImpl*>(lpDDSrcSurface)->texture, &srcRect, &dstRect);
+		SDL_RenderPresent(renderer);
```

Comment:
I think you only need to do `SDL_RenderPresent` in `IDirectDrawSurface::Flip`?
Untested, so please verify.
```suggestion
```

Reply:
No that won't work, the game does not call both. I know it seems wrong but they either `memcpy()` to the back buffer and call `->Flip()` or they just call `->Blt()` even though that looks wrong according to the docs I have looked at.